### PR TITLE
Document SHOW FUNCTIONS and SHOW PROCEDURES

### PR DESCRIPTION
**Pull request by Hunterness**
Depends on ~~https://github.com/neo-technology/neo4j/pull/9840 and https://github.com/neo-technology/neo4j/pull/9908~~

Companion to https://github.com/neo-technology/neo4j-manual-modeling/pull/1982

(I have NOT built and looked at this)

### CODE REVIEW

**Comment by mnd999** [comment]
Getting a failure in the docs build:
```
Exception in thread "main" org.neo4j.graphdb.QueryExecutionException: Unsupported administration command: SHOW ALL PRIVILEGES YIELD * WHERE action='execute' AND segment STARTS WITH $seg RETURN access, segment, collect(role) as roles
	at org.neo4j.kernel.impl.query.QueryExecutionKernelException.asUserException(QueryExecutionKernelException.java:35)
	at org.neo4j.kernel.impl.coreapi.TransactionImpl.execute(TransactionImpl.java:262)
	at org.neo4j.kernel.impl.coreapi.TransactionImpl.execute(TransactionImpl.java:243)
	at org.neo4j.cypher.internal.runtime.interpreted.commands.showcommands.ShowProcFuncCommandHelper$.getPrivileges(ShowProcFuncCommandHelper.scala:107)
	at org.neo4j.cypher.internal.runtime.interpreted.commands.showcommands.ShowProceduresCommand.originalNameRows(ShowProceduresCommand.scala:46)
	at org.neo4j.cypher.internal.runtime.interpreted.commands.showcommands.Command.rows(Command.scala:32)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.CommandPipe.internalCreateResults(CommandPipe.scala:28)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.Pipe.createResults(Pipe.scala:43)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.Pipe.createResults$(Pipe.scala:40)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.CommandPipe.createResults(CommandPipe.scala:26)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.createResults(Pipe.scala:67)
	at org.neo4j.cypher.internal.runtime.interpreted.PipeExecutionResult.request(PipeExecutionResult.scala:63)
	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.request(StandardInternalExecutionResult.scala:90)
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.request(ClosingExecutionResult.scala:144)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.fetchResults(ResultSubscriber.java:355)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.nextFromSubscriber(ResultSubscriber.java:323)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.fetchNextOrNull(ResultSubscriber.java:294)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.fetchNextOrNull(ResultSubscriber.java:58)
	at org.neo4j.internal.helpers.collection.PrefetchingIterator.peek(PrefetchingIterator.java:60)
	at org.neo4j.internal.helpers.collection.PrefetchingIterator.hasNext(PrefetchingIterator.java:46)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:132)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.neo4j.doc.ProcedureReferenceGenerator.parseResult(ProcedureReferenceGenerator.java:116)
	at org.neo4j.doc.ProcedureReferenceGenerator.procedures(ProcedureReferenceGenerator.java:108)
	at org.neo4j.doc.ProcedureReferenceGenerator.communityEditionProcedures(ProcedureReferenceGenerator.java:89)
	at org.neo4j.doc.ProcedureReferenceGenerator.document(ProcedureReferenceGenerator.java:63)
	at org.neo4j.doc.GenerateProcedureReference.main(GenerateProcedureReference.java:62)
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala [hunk 1]
```diff
@@ -379,6 +379,33 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
     )
   }
 
+  @Test def showFunctions() {profileQuery(
+      title = "Listing functions",
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala:383]: We use "List indexes" and "List constraints" but "Listing" here - we should be consistent.

File: cypher/cypher-docs/src/docs/dev/clause.asciidoc [hunk 2]
```diff
@@ -239,6 +240,22 @@ m|--- <<query-using-periodic-commit-hint, USING PERIODIC COMMIT>>
 |===
 
 
+[[header-listing-clauses]]
+**Listing clauses**
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/clause.asciidoc]: I don't really like "Listing Clauses", but I'm struggling to think of anything better. Maybe we can go more specific - "Listing Procedures and Functions", but that doesn't scale well. Metadata, and catalogue are too vague. Maybe DBMS [query] clauses? Perhaps docs have a better suggestion.

File: cypher/cypher-docs/src/docs/dev/ql/administration/constraints/index.asciidoc [hunk 3]
```diff
@@ -75,7 +75,7 @@ Using the keyword `IF NOT EXISTS` makes the command idempotent, and no error wil
 [[administration-constraints-syntax]]
 == Syntax
 
-.Syntax for managing indexes
+.Syntax for managing constraints
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/ql/administration/constraints/index.asciidoc:78]: This doesn't seem related, is it intentional?

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    p("include::listing-functions-columns.asciidoc[]")
+    section("Examples") {
+      section("Listing all functions") {
+        p(
+          """To list all available functions with the default output columns, the `SHOW FUNCTIONS` command can be used.
+            |If all columns are wanted, use `SHOW FUNCTIONS YIELD *`.""".stripMargin)
+        query("SHOW FUNCTIONS", ResultAssertions(p => {
+          p.columns should contain theSameElementsAs Array("name", "category", "description")
+        })) {
+          resultTable()
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala]: This table is very large - probably we should restrict it with LIMIT, or omit it entirely.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        p(
+          """There are two versions of the `EXECUTABLE` clause.
+            |The first one is to filter for the current user:""".stripMargin)
+        login("jake", "abc123")
+        query("SHOW FUNCTIONS EXECUTABLE BY CURRENT USER YIELD *", ResultAssertions(p => {
+          p.columns should contain theSameElementsAs Array("name", "category", "description", "signature", "isBuiltIn", "argumentDescription", "returnDescription", "aggregating", "rolesExecution", "rolesBoostedExecution")
+          p.columnAs[List[String]]("rolesExecution").foreach(_ should be(null))
+          p.columnAs[List[String]]("rolesBoostedExecution").foreach(_ should be(null))
+        })) {
+          resultTable()
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala]: Again, this is huge, please restrict the output, and lots of columns make it not-so-readable:
image

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        }
+        p("Notice that the two roles columns are empty due to missing the <<administration-security-administration-dbms-privileges-role-management, SHOW ROLE>> privilege.")
+        logout()
+        p("The second versions is to filter for a specific user:")
+        query("SHOW FUNCTIONS EXECUTABLE BY jake", ResultAssertions(p => {
+          p.columns should contain theSameElementsAs Array("name", "category", "description")
+        })) {
+          resultTable()
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala]: Again, this is a lot

File: cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+| `+`
+| `+`
+
+| `description`
+| The procedure description.
+| `+`
+| `+`
+
+| `mode`
+| The procedure mode, for example `READ` or `WRITE`.
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc:20]: Is there a reference for procedure modes we can link to from here?

File: cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+|
+| `+`
+
+| `returnDescription`
+| List of the returned values for the procedure, as map of strings with name, type, and description.
+|
+| `+`
+
+| `admin`
+| If it is or isn't an admin procedure.
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc]:
```suggestion
`true` if this procedure is an admin procedure
```

File: cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+|
+| `+`
+
+| `rolesExecution`
+| List of roles who can execute this procedure.
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc]:
```suggestion
| List of roles permitted to execute this procedure.
```

File: cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+Will be `null` without the <<administration-security-administration-dbms-privileges-role-management, SHOW ROLE>> privilege.
+|
+| `+`
+
+| `rolesBoostedExecution`
+| List of roles who will use boosted mode when executing this procedure.
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc]:
```suggestion
| List of roles permitted to use boosted mode when executing this procedure.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    p("include::listing-procedures-columns.asciidoc[]")
+    section("Examples") {
+      section("Listing all procedures") {
+        p(
+          """To list all available procedures with the default output columns, the `SHOW PROCEDURES` command can be used.
+            |If all columns are wanted, use `SHOW PROCEDURES YIELD *`.""".stripMargin)
+        query("SHOW PROCEDURES", ResultAssertions(p => {
+          p.columns should contain theSameElementsAs Array("name", "description", "mode", "worksOnSystem")
+        })) {
+          resultTable()
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]: Again, this is 88 rows, please limit somehow

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+      }
+      section("Listing procedures with filtering on output columns") {
+        p(
+          """The listed procedures can be filtered in multiple ways, one way is to use the `WHERE` clause.
+            |For example getting the name of all admin procedures:""".stripMargin)
+        query("SHOW PROCEDURES YIELD name, admin WHERE admin", ResultAssertions(p => {
+          p.columns should contain theSameElementsAs Array("name", "admin")
+          p.columnAs[Boolean]("admin").foreach(_ should be(true))
+        })) {
+          resultTable()
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]: Maybe swap the columns here would be nicer:
image

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+        p(
+          """There are two versions of the `EXECUTABLE` clause.
+            |The first one is to filter for the current user:""".stripMargin)
+        login("jake", "abc123")
+        query("SHOW PROCEDURES EXECUTABLE BY CURRENT USER YIELD *", ResultAssertions(p => {
+          p.columns should contain theSameElementsAs Array("name", "description", "mode", "worksOnSystem", "signature", "argumentDescription", "returnDescription", "admin", "rolesExecution", "rolesBoostedExecution", "option")
+          p.columnAs[List[String]]("rolesExecution").foreach(_ should be(null))
+          p.columnAs[List[String]]("rolesBoostedExecution").foreach(_ should be(null))
+        })) {
+          resultTable()
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]: Again, many narrow columns are hard to read and 57 rows is a lot:
image

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+        }
+        p("Notice that the two roles columns are empty due to missing the <<administration-security-administration-dbms-privileges-role-management, SHOW ROLE>> privilege.")
+        logout()
+        p("The second versions is to filter for a specific user:")
+        query("SHOW PROCEDURES EXECUTABLE BY jake", ResultAssertions(p => {
+          p.columns should contain theSameElementsAs Array("name", "description", "mode", "worksOnSystem")
+        })) {
+          resultTable()
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]: Again, 57 rows please restrict.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+        query("SHOW PROCEDURES EXECUTABLE BY CURRENT USER YIELD *", ResultAssertions(p => {
+          p.columns should contain theSameElementsAs Array("name", "description", "mode", "worksOnSystem", "signature", "argumentDescription", "returnDescription", "admin", "rolesExecution", "rolesBoostedExecution", "option")
+          p.columnAs[List[String]]("rolesExecution").foreach(_ should be(null))
+          p.columnAs[List[String]]("rolesBoostedExecution").foreach(_ should be(null))
+        })) {
+          resultTable()
+        }
+        p("Notice that the two roles columns are empty due to missing the <<administration-security-administration-dbms-privileges-role-management, SHOW ROLE>> privilege.")
+        logout()
+        p("The second versions is to filter for a specific user:")
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]:
```suggestion
        p("The second version filters the list to only contain procedures executable by a specific user:")
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+        }
+      }
+      section("Listing procedures with other filtering") {
+        database("system")
+        initQueries(
+          "CREATE USER jake SET PASSWORD 'abc123' CHANGE NOT REQUIRED"
+        )
+        database("neo4j")
+        p(
+          """The listed procedures can also be filtered on whether a user can execute them.
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]:
```suggestion
          """The listed procedures can also be filtered by whether a user can execute them.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+    doc("Listing functions", "query-listing-functions")
+    p("Listing the available functions can be done with `SHOW FUNCTIONS`.")
+    p("include::listing-functions-syntax.asciidoc[]")
+    p("This command will produce a table with the following columns:")
+    p("include::listing-functions-columns.asciidoc[]")
+    section("Examples") {
+      section("Listing all functions") {
+        p(
+          """To list all available functions with the default output columns, the `SHOW FUNCTIONS` command can be used.
+            |If all columns are wanted, use `SHOW FUNCTIONS YIELD *`.""".stripMargin)
```
Comment (mnd999) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala]:
```suggestion
            |If all columns are required, use `SHOW FUNCTIONS YIELD *`.""".stripMargin)
```

File: cypher/cypher-docs/src/docs/dev/ql/listing/listing-functions-columns.adoc [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+|
+| `+`
+
+| `aggregating`
+| If the function is aggregating or not.
+|
+| `+`
+
+| `rolesExecution`
+| List of roles who can execute this function.
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/ql/listing/listing-functions-columns.adoc]:
```suggestion
| List of roles permitted to execute this function.
```

File: cypher/cypher-docs/src/docs/dev/ql/listing/listing-functions-columns.adoc [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+
+Will be `null` without the <<administration-security-administration-dbms-privileges-role-management, SHOW ROLE>> privilege.
+|
+| `+`
+
+| `rolesBoostedExecution`
+| List of roles who will use boosted mode when executing this function.
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/ql/listing/listing-functions-columns.adoc]:
```suggestion
| List of roles permitted to use boosted mode when executing this function.
```

File: cypher/cypher-docs/src/docs/dev/ql/listing/listing-functions-columns.adoc [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+|
+| `+`
+
+| `returnDescription`
+| The return value type.
+|
+| `+`
+
+| `aggregating`
+| If the function is aggregating or not.
```
Comment (mnd999) [inline cypher/cypher-docs/src/docs/dev/ql/listing/listing-functions-columns.adoc]:
```suggestion
| Whether the function is aggregating or not.
```

**Review by mnd999 — CHANGES_REQUESTED** [review]

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala (hunk 1 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala:383]: But we already have `List functions` which is functions of category `List`... (I started by adding the functions to the same chapter as that hence not wanting to use the same name). 

I also don't want to risk confusing these with lists in cypher...

But I agree that we should be consistent... Not sure if it is better to update indexes/constraints but I guess the admin commands also use list (or maybe listing?)

File: cypher/cypher-docs/src/docs/dev/clause.asciidoc (hunk 2 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/clause.asciidoc]: > I don't really like "Listing Clauses", but I'm struggling to think of anything better.

That was my exact problem

File: cypher/cypher-docs/src/docs/dev/ql/administration/constraints/index.asciidoc (hunk 3 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/ql/administration/constraints/index.asciidoc:78]: I copied this tale to use as a base for the procedure/function ones and realized the table for `CONSTRAINTS` said `INDEXES` so I updated that while I noticed it

So it is intentional and technically unrelated

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala (hunk 4 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala]: It it very large but I'm unsure if I want to not display it at all. And I wanted to have the `SHOW` command without any extra clauses so I can't have limit

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala (hunk 5 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowFunctionsTest.scala]: this I could add limit to (and maybe yield specific instead of `*`)

File: cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc (hunk 7 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc:20]: maybe, but probably not, can see if I can find anything

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala (hunk 12 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]: That is a problem all over and comes from me not wanting every example to have a return clause and therefore getting the columns in alphabetical order. This one I think could have it though

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala (hunk 13 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]: I know, all of these are horrible result tables.

But I don't want to get rid of all tables and I want to have different types of queries and not just have yield and return on all of them so I'm not sure how to fix this...

I could maybe try to make some custom resultTable methods that just prints the first 10 rows and reorder the columns to the order we want. can't really do anything about the amount of columns when using `yield *` though

Comment (mnd999) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]: I think a shorter result table would be generally useful, so yeah that sounds good. Maybe it could also limit the number of columns and just show an empty column at the end with '...' as the header row. Not sure how easy that is to do though.

Comment (Hunterness) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ShowProceduresTest.scala]: Any thoughts as to how many rows would be a good cut-off?

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala (hunk 1 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala:383]: Since all administration `SHOW` commands have `Listing` I'll update indexes/constraints to use that as well for consistency

File: cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc (hunk 7 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/ql/listing/listing-procedures-columns.adoc:20]: They are described in the java reference in the section about writing [user-defined procedures](https://neo4j.com/docs/java-reference/current/extending-neo4j/procedures-and-functions/procedures/#writing-procedure), but I don't find anything else when searching the docs

**Comment by mnd999** [comment]
As discussed, one new observation, the Listings Clauses seem to be inlined on the Clauses page which leads to the menu on the RHS:

image

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala (hunk 1 above)
Comment (mnd999) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala:383]: I think you missed one:
image

QueryPlanTest, I think.

**Comment by Hunterness** [comment]
A bit more info around the listing clauses and content list: The other clauses (like `MATCH`) gets put on separate pages and linked in the left side content list. The listing clauses gets put as new sections on the same page as `Clauses` instead and therefore in the right hand side contents.

For docs team reviewer: Is there a way to get `SHOW PROCEDURE` and `SHOW FUNCTIONS` to become separate pages as well and behaving the same way as the other clauses?

**Comment by davidoliverSP2** [comment]
> For docs team reviewer: Is there a way to get `SHOW PROCEDURE` and `SHOW FUNCTIONS` to become separate pages as well and behaving the same way as the other clauses?

@Hunterness I'll have a go, and I'll push to the PR if I'm able to!

**Comment by davidoliverSP2** [comment]
> As discussed, one new observation, the Listings Clauses seem to be inlined on the Clauses page which leads to the menu on the RHS:
> 
> image

Fixed in https://github.com/neo-technology/neo4j-manual-modeling/pull/1982/commits/1c2416b17564ea8a5808d73d1de6567a74aeb438

File: cypher/cypher-docs/src/docs/dev/clause.asciidoc (hunk 2 above)
Comment (davidoliverSP2) [reply cypher/cypher-docs/src/docs/dev/clause.asciidoc]: @Hunterness and I discussed this, and thought that **Listing functions and procedures** is the best bet for now

File: cypher/cypher-docs/src/docs/dev/clause.asciidoc [hunk 21]
```diff
@@ -240,8 +240,8 @@ m|--- <<query-using-periodic-commit-hint, USING PERIODIC COMMIT>>
 |===
 
 
-[[header-listing-clauses]]
-**Listing clauses**
+[[header-listing-procs-functions]]
```
Comment (Hunterness) [inline cypher/cypher-docs/src/docs/dev/clause.asciidoc:243]: We are referencing to this in one more place

File: cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/ShowProcFuncTest.scala [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+
+import org.neo4j.cypher.docgen.RefcardTest
+import org.neo4j.cypher.docgen.tooling.DocsExecutionResult
+import org.neo4j.cypher.docgen.tooling.QueryStatisticsTestSupport
+import org.neo4j.graphdb.Transaction
+
+class ShowProcFuncTest extends RefcardTest with QueryStatisticsTestSupport {
+  val graphDescription = List()
+  val title = "SHOW FUNCTIONS and PROCEDURES"
+  override val linkId = "clauses/#header-listing-clauses"
```
Comment (Hunterness) [inline cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/ShowProcFuncTest.scala]:
```suggestion
  override val linkId = "clauses/#header-listing-procs-functions"
```

File: cypher/cypher-docs/src/docs/dev/clause.asciidoc (hunk 2 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/clause.asciidoc]: Can add the reasoning:
1. We are not sure what is coming down the pipeline so we might as well keep it explicit now and update it to be more inclusive when we know what to add.
2. The one thing we do know is coming are transaction and query commands, which will be both listing and terminating. This means they will most likely get their own section anyway (or the section needs to be updated to `management` or something to include terminate as well and then we might as well be descriptive now).

**Review by davidoliverSP2 — APPROVED** [review]
Looks good, thanks @Hunterness 

I just need to verify the refcard changes in the corresponding https://github.com/neo-technology/neo4j-manual-modeling/pull/1982
